scalameta/contrib/src/main/scala/scala/meta/contrib/instances/StatementsInstances.scala

```diff
@@ -0,0 +1,117 @@
+package scala.meta.contrib.instances
+
+import scala.collection.immutable.Seq
+import scala.meta._
+import scala.meta.contrib._
+import scala.meta.contrib.implicits.Statements
+
+trait StatementsInstances {
+  implicit val termBlockStats: Statements[Term.Block] = new Statements[Term.Block] {
+    override def getStats(block: Term.Block): Seq[Stat] = block.stats
+
+    override def replaceStats(block: Term.Block, newStats: Seq[Stat]): Term.Block =
+      block.copy(stats = newStats)
+  }
+
+  implicit val packageStats: Statements[Pkg] = new Statements[Pkg] {
+    override def getStats(pkg: Pkg): Seq[Stat] = pkg.stats
+
+    // TODO: Packages have restrictions that all others dont.
+    // aka top level stats only
+    override def replaceStats(pkg: Pkg, newStats: Seq[Stat]): Pkg =
+      pkg.copy(stats = newStats)
+  }
+
+  implicit val templStats: Statements[Template] = new Statements[Template] {
+    override def getStats(templ: Template): Seq[Stat] =
+      templ.stats match {
+        case Some(stats) => stats
+        case None => Nil
+      }
+
+    override def replaceStats(templ: Template, newStats: Seq[Stat]): Template = {
+      newStats match {
+        case Nil => templ.copy(stats = None)
+        case s => templ.copy(stats = Some(s))
+      }
+    }
+  }
+
+  implicit val classStats: Statements[Defn.Class] = new Statements[Defn.Class] {
+    override def getStats(clazz: Defn.Class): Seq[Stat] =
+      clazz.templ.getStats
+
+    override def replaceStats(clazz: Defn.Class, newStats: Seq[Stat]): Defn.Class = {
+      clazz.copy(templ = clazz.templ.replaceStats(newStats))
+    }
+  }
+
+  implicit val objectStats: Statements[Defn.Object] = new Statements[Defn.Object] {
+    override def getStats(obj: Defn.Object): Seq[Stat] =
+      obj.templ.getStats
+
+    override def replaceStats(obj: Defn.Object, newStats: Seq[Stat]): Defn.Object =
+      obj.copy(templ = obj.templ.replaceStats(newStats))
+  }
```

Comment:
withStats?  
This looks like sugar to prevent needing to do the copy repeatedly.  What about preventing the need for getStats and copy by creating withMappedStats that takes a function from Seq[Stat] => Seq[Stat]?